### DISCUSSION TITLE
Where does the global view helper need to be placed?

### DISCUSSION THREAD

**Discussion by zw963** [luckyframework/lucky#1888] [Q&A] [answered]
Create a new helper method which can be used in all view file.

For now, i am doing like this:

1. create a new layout, e.g. pages/base_layout.cr

```crystal
abstract class BaseLayout
  include Lucky::HTMLPage

  def render
    content
  end

  private def helper_foo
    # ...
  end
end
```

Then inherit all others page from it.

```crystal
abstract class AuthLayout < BaseLayout
  # ...
end
```

```crystal
abstract class MainLayout < BaseLayout
  # ...
end
```

I have to change src/app.cr to `require base_layout` before others

```diff
# ...
+ require "./pages/base_layout"
require "./pages/**"
```

Then, i can use helper_foo every where as expected.

My question is, is this the expected usage? 

If is is yes, should we create `base_layout` equivalent when create a new repo?
If it is no, what is the prefer way to do that?

Thanks

**Comment by jwoertink** [comment, marked as answer]
You could either put this in `src/components/shared` or make a `src/pages/mixins`. 

```crystal
abstract class BaseLayout
  include Lucky::HTMLPage
  include PageHelpers

  def render
    content
  end
end

# src/pages/mixins
module PageHelpers
  def helper_foo

  end
end
```

Just be sure to require the new folder before your layouts.

**Reply by zw963** [reply]
Thanks, this is better than class inherit, i can even use view helper in action if included.
